pkg/actuators/machine/actuator_test.go

```diff
@@ -32,6 +41,50 @@ func TestMachineEvents(t *testing.T) {
 	g := NewWithT(t)
 	ctx := context.Background()
 
+	// Create the Infrastructure CR
+	infra := &configv1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: globalInfrastuctureName,
+		},
+		Spec: configv1.InfrastructureSpec{
+			CloudConfig: configv1.ConfigMapFileReference{},
+			PlatformSpec: configv1.PlatformSpec{
+				Type: configv1.NutanixPlatformType,
+				Nutanix: &configv1.NutanixPlatformSpec{
+					PrismCentral: configv1.NutanixPrismEndpoint{Address: "10.40.142.15", Port: 9440},
+					PrismElements: []configv1.NutanixPrismElementEndpoint{
+						{Name: "ganon", Endpoint: configv1.NutanixPrismEndpoint{Address: "10.40.231.131", Port: 9440}},
+					},
+				},
+			},
+		},
+	}
+
+	g.Expect(k8sClient.Create(ctx, infra)).To(Succeed())
+	defer func() {
+		g.Expect(k8sClient.Delete(ctx, infra)).To(Succeed())
+	}()
+	g.Expect(strings.EqualFold(infra.Spec.PlatformSpec.Nutanix.PrismCentral.Address, "10.40.142.15")).Should(BeTrue())
+	g.Expect(infra.Spec.PlatformSpec.Nutanix.PrismCentral.Port == 9440).Should(BeTrue())
+	g.Expect(len(infra.Spec.PlatformSpec.Nutanix.PrismElements) == 1).Should(BeTrue())
+	g.Expect(infra.Spec.PlatformSpec.Nutanix.PrismElements[0].Name == "ganon").Should(BeTrue())
+	g.Expect(infra.Spec.PlatformSpec.Nutanix.PrismElements[0].Endpoint.Address == "10.40.231.131").Should(BeTrue())
+	g.Expect(infra.Spec.PlatformSpec.Nutanix.PrismElements[0].Endpoint.Port == 9440).Should(BeTrue())
```

Comment:
Do we need to perform these checks given we just set the fields above?

Reply:
This is to ensure that the PC/PE endpoints set in the Infrastructure CR are kept after the CR is created.

Reply:
Ok, i don't have a strong opinion either way on whether to keep these or not, but if you do want to have them, can I suggest rather than checking everything is true, to use the assertion matchers. If you do it like this and any of them fail, you will have a result printed that shows the difference, otherwise currently you'll get `expected false to be true` which isn't great

```suggestion
	g.Expect(infra.Spec.PlatformSpec.Nutanix.PrismCentral.Address).Should(Equal("10.40.142.15"))
	g.Expect(infra.Spec.PlatformSpec.Nutanix.PrismCentral.Port).Should(Equal(9440))
	g.Expect(infra.Spec.PlatformSpec.Nutanix.PrismElements).Should(HaveLen(1))
	g.Expect(infra.Spec.PlatformSpec.Nutanix.PrismElements[0].Name).Should(Equal("ganon"))
	g.Expect(infra.Spec.PlatformSpec.Nutanix.PrismElements[0].Endpoint.Address).Should(Equal("10.40.231.131"))
	g.Expect(infra.Spec.PlatformSpec.Nutanix.PrismElements[0].Endpoint.Port).Should(Equal(9440))
```

Reply:
Just talking to @elmiko about this, If you are keeping these tests, please apply my suggestion so that we have the correct output should any of them fail.

Reply:
Changed as suggested.